### ISSUE TITLE
Combobox item selection state is not read back to screen reader users

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#4917]
### Summary

Unselected states for `combobox-item`s are read back to screen reader users, but selected ones are not. 

**Some of the expected behavior has been mitigated, but additional development is needed.**

### Actual Behavior

If a user selects a `combobox-item`, context is not currently provided to the screen reader that selection has taken place. When (Unselected items are read back as unselected.)

Also, if the user returns to the input, currently context is not provided to the screen reader that an item has been previously selected.

### Expected Behavior

- [x] When selecting a `combobox-item`, screen reader users should be provided context that the item was selected.
- [ ] When navigating the `combobox-item`s, screen reader users should be provided context that the item has been selected.
- [ ] When selections have been made in the `combobox` the number of items selected should be read back to screen readers (e.g., "1 item selected")
    - [Sample: Accessible Multi-Selection Dropdown](https://codepen.io/geospatialem/pen/oNpVLYx) 

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/combobox/#sample

### Reproduction Steps

1. Activate a screen readers (JAWS/NVDA/VoiceOver)
2. Navigate to the [sample](https://developers.arcgis.com/calcite-design-system/components/combobox/#sample)
3. Keyboard navigate the `combobox` and select an item in the list using the arrow keys for navigation and the `Enter` key for selection
    - Observe the behavior of non-selected items (that are provided context)
    - Observe the behavior upon selecting (provided no context)
    - Navigate away from the selected item, and back to it using the arrows keys to observe the behavior of the selected item (provided no context)

### Reproduction Version

beta.83

### Relevant Info

Present across screen readers (JAWS, NVDA, VoiceOver)

cc #4388

**Comment by driskull** [comment]
The combobox already sets `aria-selected` on the item. Is there something missing?

**Comment by geospatialem** [comment]
The second and third bullet points above are likely a result of assistive technology (AT) products. Since there is still some context provided, we are meeting the minimum requirements. Closing this issue for now, and can re-evaluate if a new issue is created in the future.

> 2. When navigating the combobox-items, screen reader users should be provided context that the item has been selected.
 
> 3. When selections have been made in the combobox the number of items selected should be read back to screen readers (e.g., "1 item selected")

cc @SaraLawrence
